src/sdi-snap.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "sdi-snap.h"
+
+enum { PROP_NAME = 1, PROP_DIALOG, PROP_LAST };
+
+struct _SdiSnap {
+  GObject parent_instance;
+
+  gchar *name;
+  SdiRefreshDialog *dialog;
+  gboolean ignored;
+  gboolean inhibited;
+  gboolean hiden;
+  gboolean manually_hiden;
+  gint destroy_id;
+};
+
+G_DEFINE_TYPE(SdiSnap, sdi_snap, G_TYPE_OBJECT)
+
+SdiRefreshDialog *sdi_snap_get_dialog(SdiSnap *self) {
+  g_return_val_if_fail(SDI_IS_SNAP(self), NULL);
+
+  if (self->dialog == NULL)
+    return NULL;
+  else
+    return g_object_ref_sink(self->dialog);
```

Comment:
Does this need to be a ref_sink here - dialog is only set to a full reference in sdi_snap_set_dialog?

Reply:
I had a problem with the unsinken references (the dialogues got freed before than they should), so I decided to use ref_sink everywhere just to avoid problems... But maybe it's better to just return a sinked reference in the constructor.